src/content/advanced/iam-roles-for-service-accounts/index.md

```diff
@@ -0,0 +1,105 @@
+
+Make sure the resource has the `alpha.aws.giantswarm.io/iam-roles-for-service-accounts` annotation. The value can be anything you like, as only the presence of that annotation is checked. Here is an example:
+
+```yaml
+apiVersion: infrastructure.giantswarm.io/v1alpha3
+kind: AWSCluster
+metadata:
+  annotations:
+    alpha.aws.giantswarm.io/iam-roles-for-service-accounts: ""
+  labels:
+    aws-operator.giantswarm.io/version: 10.18.0
+    cluster.x-k8s.io/cluster-name: abcl0
+    giantswarm.io/cluster: abcl0
+    giantswarm.io/organization: myorg
+    release.giantswarm.io/version: 17.1.0
+  name: abcl0
+  namespace: myorg
+spec:
+  ...
+```
+
+Alternatively you can use `kubectl` command to annotate the CR like shown below:
+
+```bash
+kubectl annotate awsclusters.infrastructure.giantswarm.io abcl0 alpha.aws.giantswarm.io/iam-roles-for-service-accounts=""
+```
+
+In order to apply the changes, rolling of the master nodes is required. Rolling of the nodes can be triggered either by an update or by manually by terminating each node.
+
+## Using IAM roles for service accounts
+
+### IAM Role
+In oder to use the IAM role with a service account you need to create new AWS role and configure `Trusted entities` with folowing statement:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn:aws:iam::$AWS_ACCOUNT:oidc-provider/s3-eu-west-1.amazonaws.com/$AWS_ACCOUNT-g8s-$CLUSTER_ID-oidc-pod-identity"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "s3-eu-west-1.amazonaws.com/$AWS_ACCOUNT-g8s-$CLUSTER_ID-oidc-pod-identity:sub": "system:serviceaccount:$NAMESPACE:$SA_NAME"
+                }
+            }
+        }
+    ]
+}
+```
+
+You need to fill real values for these placeholders:
+* `$AWS_ACCOUNT` - aws account id, where is the cluster running
+* `$CLUSTER_ID` - cluster id
+* `$NAMESPACE` - Kubernetes namespace in cluster, where the pod and service account be used.
+* `$SA_NAME` - Name of the Kubernetes Service Account which wil be using by the pod.
+
+### Service account
+The service account has to be annotated with a full ARN of the IAM role, you can get the ARN of the role in AWS console, when you check role details.
+The annotation key has to be `eks.amazonaws.com/role-arn`.
+
+```
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    eks.amazonaws.com/role-arn: arn:aws:iam::$AWS_ACCOUNT:role/$IAM_ROLE_NAME
+  name: s3-access
+  namespace: default
+```
+
+## Verify your configuration is correct
+Once your pod is running with the configured service account, you should see a file in the pod called `/var/run/secrets/eks.amazonaws.com/serviceaccount/token`, which containts a JWT token with details of the role. 
```

Comment:
How can a user see this file? Is there a file mount in the pod manifest?

Reply:
you can see it on the pod manifest and also if you exec into the pod